apps/proxy/src/app/create-project/create-project.component.ts

```diff
@@ -0,0 +1,208 @@
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, Validators, FormBuilder, FormArray } from '@angular/forms';
+import { CAMPAIGN_NAME_REGEX, ONLY_INTEGER_REGEX, URL_REGEX } from '@proxy/regex';
+import { CustomValidators } from '@proxy/custom-validator';
+import { environment } from '../../environments/environment';
+import { CreateProjectComponentStore } from './create-project.store';
+import { Observable, takeUntil } from 'rxjs';
+import { IPaginatedResponse } from '@proxy/models/root-models';
+import { IEnvironments, IProjects } from '@proxy/models/logs-models';
+import { BaseComponent } from '@proxy/ui/base-component';
+import { select, Store } from '@ngrx/store';
+import { IAppState, selectAllProjectList } from '../ngrx';
+import { rootActions } from '../ngrx/actions';
+
+@Component({
+    selector: 'proxy-create-project',
+    templateUrl: './create-project.component.html',
+    styleUrls: ['./create-project.component.scss'],
+    providers: [CreateProjectComponentStore],
+})
+export class CreateProjectComponent extends BaseComponent implements OnInit {
+    public primaryDetailsForm: FormGroup;
+    public gatewayUrlDetailsForm: FormGroup;
+    public destinationUrlForm: FormGroup;
+    public currentstep: number = 1;
+    public checked: Boolean = false;
+    public showEndpoint: Boolean = false;
+    public environments_with_slug;
+    public environmentParams = {
+        itemsPerPage: 5,
+        pageNo: 1,
+    };
+    public projects$: Observable<IPaginatedResponse<IProjects[]>>;
+    public environments$: Observable<IPaginatedResponse<IEnvironments[]>> = this.componentStore.environments$;
+    public sourceDomain$: Observable<any> = this.componentStore.sourceDomain$;
+    public getProject$: Observable<boolean> = this.componentStore.getProject$;
+    public isLoading$: Observable<boolean> = this.componentStore.isLoading$;
+    public projectId: number;
+
+    constructor(
+        private componentStore: CreateProjectComponentStore,
+        private fb: FormBuilder,
+        private store: Store<IAppState>
+    ) {
+        super();
+        this.projects$ = this.store.pipe(select(selectAllProjectList));
+        this.primaryDetailsForm = this.fb.group({
+            name: [
+                '',
+                [
+                    Validators.required,
+                    Validators.pattern(CAMPAIGN_NAME_REGEX),
+                    CustomValidators.minLengthThreeWithoutSpace,
+                    CustomValidators.noStartEndSpaces,
+                    Validators.maxLength(20),
+                ],
+            ],
+            rateLimitHit: [null, [Validators.required, Validators.pattern(ONLY_INTEGER_REGEX)]],
+            rateLimitMinute: [null, [Validators.required, Validators.pattern(ONLY_INTEGER_REGEX)]],
+            selectedEnvironments: [''],
+        });
+
+        (this.destinationUrlForm = this.fb.group({
```

Comment:
why wrapped in brackets it can simply done like this
 this.destinationUrlForm = this.fb.group({
            endpoint: ['', Validators.required],
            ForwardUrl: this.fb.array([]),
        });
        this.gatewayUrlDetailsForm = this.fb.group({
            useSameUrlForAll: [false],
            gatewayUrls: this.fb.array([]),
        });